orders/models.py

```diff
@@ -71,3 +72,56 @@ def cancel_order(self):
 
     def get_order_total_price(self):
         return self.price_per_day * (self.end_date - self.start_date).days
+
+    def get_order_status(self):
+        if self.status == StatusOptions.Pending:
+            return "Pending"
+        elif self.status == StatusOptions.Approved:
+            return "Approved"
+        elif self.status == StatusOptions.Canceled:
+            return "Canceled"
+        elif self.status == StatusOptions.OnGoing:
+            return "On going"
+        return "Finished"
+
+    @staticmethod
+    def get_capacity_of_daycare_in_dates_range(daycare_id, start_date, end_date):
+        capacity_per_day_list = [0] * ((end_date - start_date).days + 2)
+        relevant_orders = Order.objects.filter(daycare_id=daycare_id, status__in=['A', 'O'])
+        start_date = datetime.date(year=start_date.year, month=start_date.month, day=start_date.day)
+        end_date = datetime.date(year=end_date.year, month=end_date.month, day=end_date.day)
+
+        for order in relevant_orders:
+            if end_date < order.start_date or order.end_date < start_date:
+                continue
+
+            number_of_days = (order.end_date - order.start_date).days
+
+            for day in range(number_of_days):
+                current_date = order.start_date + datetime.timedelta(days=day)
+                if current_date < start_date:
+                    continue
+                elif current_date > end_date:
+                    break
+                else:
+                    capacity_per_day_list[day] = capacity_per_day_list[day] + 1
+
+        return capacity_per_day_list
+
+    def are_order_dates_are_not_occupied_for_approval(self):
+        capacity_per_day_list = self.get_capacity_of_daycare_in_dates_range(self.daycare_id,
+                                                                            self.start_date,
+                                                                            self.end_date)
+
+        daycare_capacity = DayCare.objects.get(name=self.daycare_id.name).capacity
+        return all(current_capacity < daycare_capacity for current_capacity in capacity_per_day_list)
+
+    def is_the_order_cancelable(self):
+        if self.status == StatusOptions.Pending or self.status == StatusOptions.Approved:
```

Comment:
Instead of the `if`, you can return the result of the conditional.

Reply:
You're right. Changed.